TESTARtool/TESTAR_dev
# Webdriver: improvement, issues solving and small refactoring. General: Paste Action and others

TESTOMAT, DECODER, iv4XR, IVVES : improvements and issues solving

New Features to test

1. Webdriver development:

- Add Webdriver StateModel Management Tags (settings and GUI level)
- Refactor and move "settings/webdriver_protocols" generic utility content. Utility webdriver methods like "detectForcedActions", "detectForcedPopupClick", etc... have been moved to "WebdriverProtocol.java".
- Shadow Web Elements should be detected now
- Avoid kill process exception with SPY + Webdriver, do not call killTestLaunchedProcesses
- Improve scrolling actions comparing document body bounds with window bounds

2. Create PasteText Action

- Create "waitLeftClickAndPasteIntoWidget" utility method
- Update examples: protocol_web_one_drive & protocol_webdriver_parabank

3. Gradle compilation tasks (more in Readme)
- gradle build, only compile at Java level
- gradle installDist and distZip, copy existing windows.dll, not using VisualStudio Tools
- gradle windowsDistribution, use VisualStudio Tools to compile a new windows distribution

4. Escape Special Strings in Action Desc and HTML report
5. Spy Mode Visual Panel will not be created over the y-axis of the screen + Add Refresh Time setting
6. Solve issue saving ESC actions (Empty State) in the State Model
